`kamel run` dry run may panic when no cluster

```
./kamel run /tmp/my-route.yaml --config configmap:my-cm -t container.limit-cpu=0.1 -t container.limit-memory=256Mi -o yaml
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x128 pc=0x1a2fee3]

goroutine 1 [running]:
github.com/apache/camel-k/v2/pkg/util/kubernetes.LookupConfigmap({0x282fd90, 0xc000a26bc0}, {0x0, 0x0}, {0x0, 0x0}, {0xc0008d063a, 0x5})
	github.com/apache/camel-k/v2/pkg/util/kubernetes/lookup.go:47 +0x1a3
```

--- a/pkg/cmd/run.go
+++ b/pkg/cmd/run.go
@@ -747,8 +747,10 @@ func (o *runCmdOptions) parseAndConvertToTrait(cmd *cobra.Command,
 		if err != nil {
 			return err
 		}
-		if err := parseConfig(o.Context, cmd, c, config, integration); err != nil {
-			return err
+		if o.OutputFormat == "" {
+			if err := parseConfig(o.Context, cmd, c, config, integration); err != nil {
+				return err
+			}
 		}
 		o.Traits = append(o.Traits, convertToTrait(convert(config), traitParam))
 	}

